### PR TITLE
Add associations

### PR DESCRIPTION
**Pull request by keebot-png**
## The following has been completed in line with project requirements:
- [x] Finished creating the remaining classes: `Rental`, `Book`, `Classroom` for school library and created the associations between them.
- [x] Made sure that there are no linter errors.
- [x] Make sure that you used correct Gitflow.
- [x] Follow list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_reader :book, :person
+  attr_accessor :date
```
Comment (Deepakdanger) [inline rental.rb]: - [ ] attr_accesor should contain all three `:date, :book, :person`.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rental
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
```
Comment (Deepakdanger) [inline book.rb:9]: - [ ] Kindly add a function for add rentals.
```
def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
   attr_accessor :name, :age
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = object_id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rental = []
     super()
   end
```
Comment (Deepakdanger) [inline person.rb:15]: - [ ] KIndly add the function to add rentals.
```ruby
 def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @keebot-png , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @keebot-png👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**